servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/SingleConcatWithPublisher.java

```diff
@@ -142,7 +144,7 @@ private void onErrorPropagateCancel(Throwable t) {
 
         @Override
         public final void onComplete() {
-            if (propagateCancel) {
+            if (propagateCancel || !deferSubscribe()) {
```

Comment:
I wasn't expecting the changes described in this PR to be made here. for example I was expecting the `SingleSubscriber.onSuccess(..)` methods such that we subscribe eve when completed if `CANCELLED` (perhaps conditionally?)

thing to watch out for is if `propagateCancel` we may subscribe to both sources concurrently, and `mayBeResultUpdater` is used to ensure we only terminate a single time. 

can we make the termination probation (on error / on complete) is consistent.